src/main/java/hudson/matrix/MatrixProject.java

```diff
@@ -1083,6 +1084,10 @@ public List<AxisDescriptor> getAxisDescriptors() {
             return r;
         }
 
+        public FormValidation doCheckDisplayName(@QueryParameter String value, @QueryParameter String name) {
+            return Jenkins.get().doCheckDisplayName(value, name);
```

Comment:
There are some limitations in `checkDependsOn`, and one of the ways around it was to proxy call this. I'm open to suggestions if anyone has an idea how to do it cleaner.